core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -912,10 +911,8 @@ class RemoteLogRetentionHandler {
 
             private final Optional<RetentionSizeData> retentionSizeData;
             private final Optional<RetentionTimeData> retentionTimeData;
-
             private long remainingBreachedSize;
-
-            private OptionalLong logStartOffset = OptionalLong.empty();
+            private Optional<Long> logStartOffset = Optional.empty();
```

Comment:
We can continue using `OptionalLong` as discussed offline as it has the required methods.